src/_pytest/capture.py

```diff
@@ -77,19 +81,29 @@ def __init__(self, method):
         self._method = method
         self._global_capturing = None
         self._current_item = None
+        self._atexit_funcs = []  # type: List[Callable]
+        atexit.register(self._atexit_run)
+        self._tmpfiles = {}
 
     def __repr__(self):
         return "<CaptureManager _method={!r} _global_capturing={!r} _current_item={!r}>".format(
             self._method, self._global_capturing, self._current_item
         )
 
+    def _atexit_register(self, func):
+        self._atexit_funcs.append(func)
+
+    def _atexit_run(self):
+        for func in self._atexit_funcs:
+            func()
+
     def _getcapture(self, method):
         if method == "fd":
-            return MultiCapture(out=True, err=True, Capture=FDCapture)
+            return MultiCapture(out=True, err=True, Capture=FDCapture, capman=self)
```

Comment:
Here we would pass `self._tmpfile_pool` to the `MultiCapture` objects.

Reply:
I think passing `capman` is more universal.
(IIRC I am using this when auto-suspending on readin from stdin)

Reply:
more universal means more coupling here... I prefer to pass only the necessary. I also suggest making it mandatory, so we don't have to check for its existense first (might require a few test changes I suppose).